### ISSUE TITLE
[vLLM] logprobs_tensors always None in vllm_tt Sampler — logprobs requests silently return no data

### ISSUE THREAD

**Issue by kmabeeTT** [tenstorrent/tt-xla#3366]
## Issue

Requesting `logprobs` via `vllm.SamplingParams(logprobs=N)` silently returns no logprob data on TT hardware. `Sampler.forward()` unconditionally returns `logprobs_tensors=None` regardless of whether the request asked for logprobs:

```python
# vllm_tt/sampler.py
def forward(self, logits, sampling_metadata):
    # TODO(houseroad): Add support for logprobs_mode.
    sampled = self.sample(logits, sampling_metadata)
    return SamplerOutput(
        sampled_token_ids=sampled.unsqueeze(-1),
        logprobs_tensors=None,  # ← always None
    )
```

`test_logprobs` in `test_sampling_params.py` is currently `@pytest.mark.skip` due to a separate numpy serialization crash (see #3310). Once that is resolved, this issue will cause the test to fail on the `assert result.logprobs is not None` assertion.

## Root Cause

The infrastructure to compute and gather logprobs already exists and is partially wired:

- `Sampler.compute_logprobs` computes log-softmax over logits
- `Sampler.gather_logprobs` gathers top-k logprobs and the sampled token rank
- `model_runner.py` calls `self.gather_logprobs(logits, selected_token_ids)` when `tpu_sampling_metadata.logprobs` is True and assembles the result into `LogprobsLists`
- `_precompile_gather_logprobs` warms up the `gather_logprobs` path at startup

The missing piece is that `Sampler.forward()` is not consulted for logprobs at all — the logprobs assembly happens in `model_runner.py` directly, bypassing `forward()`. The `logprobs_tensors=None` in `SamplerOutput` is therefore a stub left over from the TPU upstream fork. The `TODO(houseroad)` comment refers to `logprobs_mode` (a future extension for controlling which logprobs to return), not to basic logprob support.

## What Needs to Happen

1. Verify that the existing `model_runner.py` logprobs path (`gather_logprobs` → `LogprobsLists`) is correctly serialized into the `SamplerOutput` returned to the vLLM engine.

2. Verify XLA correctness of `gather_logprobs` under compilation — specifically `torch.topk`, `torch.gather`, and the rank computation:

```python
topk_logprobs, topk_indices = torch.topk(logprobs, num_logprobs, dim=-1)
token_logprobs = logprobs.gather(-1, token_ids)
token_ranks = (logprobs >= token_logprobs).sum(-1)
```

Note: `gather_logprobs` in `model_runner.py` has its `@torch.compile(backend="tt")` decorator commented out — it may currently run via XLA lazy dispatch rather than explicit compilation. Whether this produces correct results on device needs to be confirmed (IR dump + output comparison against CPU reference).

## Repro

```python
# Currently @pytest.mark.skip due to #3310 (numpy crash).
# Once #3310 is resolved, this will fail:
params = vllm.SamplingParams(temperature=0.8, logprobs=5, max_tokens=8)
result = llm.generate(prompt, params)[0].outputs[0]
assert result.logprobs is not None   # fails — always None
assert len(result.logprobs) > 0
```

## Fix Order

Resolve #3310 (numpy serialization crash) first — it kills the engine before logprobs can be exercised. Once the engine stays alive, validate the existing `model_runner.py` logprobs assembly path and confirm `gather_logprobs` produces correct output on device.
